types/time.go

```diff
@@ -2378,18 +2378,18 @@ func microSeconds(t *MysqlTime, input string, ctx map[string]int) (string, bool)
 	return input[length:], true
 }
 
-func yearNumericFourDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
-	return yearNumericNDigits(t, input, ctx, 4)
+func yearNumericFourDigits(t *MysqlTime, sc *stmtctx.StatementContext, input string, rawDate string, ctx map[string]int) (string, bool) {
+	return yearNumericNDigits(t, sc, input, rawDate, ctx, 4)
 }
 
-func yearNumericTwoDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
-	return yearNumericNDigits(t, input, ctx, 2)
+func yearNumericTwoDigits(t *MysqlTime, sc *stmtctx.StatementContext, input string, rawDate string, ctx map[string]int) (string, bool) {
+	return yearNumericNDigits(t, sc, input, rawDate, ctx, 2)
 }
 
-func yearNumericNDigits(t *MysqlTime, input string, ctx map[string]int, n int) (string, bool) {
+func yearNumericNDigits(t *MysqlTime, sc *stmtctx.StatementContext, input string, rawDate string, ctx map[string]int, n int) (string, bool) {
 	effectiveCount, effectiveValue := 0, 0
-	for effectiveCount+1 <= n {
-		value, succeed := parseDigits(input, effectiveCount+1)
+	for effectiveCount + 1 <= n {
+		value, succeed := parseDigits(input, effectiveCount + 1)
```

Comment:
```suggestion
		value, succeed := parseDigits(input, effectiveCount+1)
```

Reply:
Thank you for your review. I've fixed the mentioned problem.